Lib/Option.hpp

```diff
@@ -45,7 +45,7 @@ struct MaybeUninit {
   { return MV(_elem.init); }                                                                                  \
                                                                                                               \
   void init(T REF content)                                                                                    \
-  { ::new(&_elem)T(MV(content)); }                                                                            \
+  { ::new(&_elem.init)T(MV(content)); }                                                                       \
```

Comment:
Same as with `Coproduct` - I think we want to assign to the union field, not the whole thing.